app/controllers/boxes_controller.rb

```diff
@@ -22,9 +23,15 @@ def show
   end
 
   def inventory
-    return unless authorize_resource(@box, READ_BOX)
+    if !is_sender
+      return unless authorize_resource(@box, READ_BOX)
+    end
 
-    @samples = load_box_samples
+    @samples = load_box_samples_print
+
+    if @box.blinded and !is_sender
```

Comment:
Multiple calls to `is_sender` result in multiple data base queries. The result should be cached in a local variable or instance variable and re-used.